src/NuGet.Core/NuGet.Protocol/Converters/NuGetFrameworkConverter.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Xml.Linq;
+using Newtonsoft.Json;
+using NuGet.Frameworks;
+
+namespace NuGet.Protocol.Converters
+{
+    public class NuGetFrameworkConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            serializer.Serialize(writer, value.ToString());
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var value = serializer.Deserialize<string>(reader);
```

Comment:
```suggestion
            var value = reader.ReadString();
```

or something similar. Maybe you need to get string value, then call `Read()` to move to the next token.

Going through `JsonSerializer` will add additional CPU overhead that's not necessary (even worse if it causes additional allocations).